### ISSUE TITLE
How to generate MDD samples in the correct order

### ISSUE THREAD

**Issue by ghaddarAbs** [Alue-Benchmark/alue_baselines#3]
Hi, 

Thanks for your great work!!!!

I have an issue producing the same exact MDD data as [your notebook](https://github.com/Alue-Benchmark/alue_baselines/blob/master/notebooks/USE/Madar-Task-1.ipynb).

The original data from [Camel](https://docs.google.com/forms/d/e/1FAIpQLSfQqhxslVSkBN5ScQ2bvvM0xUVCUnjXxtvkAjupvxm3SSeZGw/viewform) are stored in 26 `.tsv` file as follow: `MADAR.corpus.{Dialect}.tsv`

The format of the files are as follow:

```
sentID.BTEC	split	lang	sent
5	corpus-6-test-corpus-26-train	ALX	هو هناك، قدام الاستعلامات السياحية على طول.
9	corpus-6-test-corpus-26-train	ALX	عمرى ما سمعت عن العنوان دة هنا.
11	corpus-6-test-corpus-26-train	ALX	امشى على طول لحد ما تشوف صيدلية.
26	corpus-6-test-corpus-26-train	ALX	بكام الفطار؟
``` 

I used the last part of the `split` column (aka. corpus-6-test-corpus-26-{train} -> train) to generate `MADAR-Corpus-26-{train,dev,test}.tsv` and I was able to get 46k/5k/5k train, dev, test samples respectively.

However, I cannot get test samples **in the same order** as you showed in [your notebook](https://github.com/Alue-Benchmark/alue_baselines/blob/master/notebooks/USE/Madar-Task-1.ipynb). For instance, this is the order that I get when I use `sentID.BTEC` (although they are duplicated) to sort the samples:

```
train
(5, 'هو هنيك، قدام معلومات السياح بالضبط.', 'ALE')
(5, 'راه تما، بالضبط قبالت مكتب المعلومات ديال السياحة.', 'FES')
(5, 'اوني غادي، قدام متاع معلومات السواح بالضبط.', 'TRI')
(5, 'هناك ، أمام بيانات السائح تماما .', 'MSA')
(5, 'ده قدامك هناك، يادوبك قدام مكتب استعلامات السياحة .', 'CAI')

dev
(398, 'بالمناسبة، اسمي هيروشي إيجيما.', 'ALE')
(398, 'بالمناسبة، سميتي هيروشي إيجيما.', 'FES')
(398, 'على فكرة، اسمي هيروشي ايجيما.', 'TRI')
(398, 'بالمناسبة ، اسمي هيروش إيجيما .', 'MSA')
(398, 'بالمناسبه، اسمي هيوشي ايجيما .', 'CAI')

test
(120, 'كنت بشتري هاد لو كنت محلك.', 'ALE')
(120, 'كون كنت فبلاستك نشري هادا.', 'FES')
(120, 'أني حنشري هادي لو كنت مكانك.', 'TRI')
(120, 'كنت سأشتري هذا إن كنت في مكانك ؟', 'MSA')
(120, 'لو كنت مكانك كنت صدقت.', 'CAI')

```
As you can see, this order is different from the one that you have in  [your notebook](https://github.com/Alue-Benchmark/alue_baselines/blob/master/notebooks/USE/Madar-Task-1.ipynb),  I think it's okay for train and dev, but not for test.

I need the correct order for the test to be able to submit to the leaderboard. 
Cloud you please share your code to convert the [original MDD data](https://docs.google.com/forms/d/e/1FAIpQLSfQqhxslVSkBN5ScQ2bvvM0xUVCUnjXxtvkAjupvxm3SSeZGw/viewform) to the same order in your files MADAR-Corpus-26-{train|dev|test}.tsv

Thanks,

**Comment by ghaddarAbs** [comment]
Here is the code I used to read the original data:

```
data_dir = "../MADAR.Parallel-Corpora-Public-Version1.1-25MAR2021/data-v1.1.public/"

bench_dict = defaultdict(list)
for filename in os.listdir(data_dir):
   if not filename.endswith(".tsv"): continue
   if filename.count(".") != 3: continue
   df = pd.read_csv(os.path.join(data_dir, filename), sep="\t")

   for idx, split, lang, s in zip(df["sentID.BTEC"], df["split"], df["lang"], df["sent"]):
       if split.count("-") != 5: continue
       split = split.split("-")[-1]
       bench_dict[split].append((idx, s, lang))

for portion, lst in bench_dict.items():
   print(portion)
   for item in sorted(lst, key= lambda x:(int(x[0])))[:5]:
       print(item)
```

**Comment by hseelawi** [comment]
Greetings @ghaddarAbs,

Thanks for your feedback. It seems that the original authors of the dataset have changed its structure. Luckily, we still have the original files. I just need to ask their permission to forward these to our participant until we find a better fix. I have already sent them and hopefully they will reply soon. If they answer positively I will send you the original files right away. Otherwise, we will work on a quick fix to this problem.

Kind regards,
Haitham

**Comment by ghaddarAbs** [comment]
Thanks for your reply @hseelawi

A fast solution can be to provide `test_index.txt`, where each line contains ("sentID.BTEC", "lang") in the correct order of your current test set.

For instance the equivalent of:
```
test
(120, 'كنت بشتري هاد لو كنت محلك.', 'ALE')
(120, 'كون كنت فبلاستك نشري هادا.', 'FES')
(120, 'أني حنشري هادي لو كنت مكانك.', 'TRI')
(120, 'كنت سأشتري هذا إن كنت في مكانك ؟', 'MSA')
(120, 'لو كنت مكانك كنت صدقت.', 'CAI')
```
can be a text file in this order:

```
120  ALE
120 FES
120  TRI
120 MSA
120 CAI
```


Since you still have the original file you can easily produce the `test_index.txt` by matching the (lang, sent) of the original file with ((lang, sent):  sentID.BTEC) of the new files.

Best regards, 
Abbas

**Comment by hseelawi** [comment]
That would in principle work but since we have to abide by the licence terms of the original authors we should first obtain some form of authorization. This might take a while but nonetheless, Dr. Habash has agreed that we send you the original data for now. I will send it over via the google group thread for this issue. Also, kindly notice that I have caught a bug in the DataGenerator class in the notebook. It seems that we are using an outdated version of it. I will be updating this very soon. Please feel free to reply or close this issue.

Kind regards,
Haitham

**Comment by ghaddarAbs** [comment]
Thanks alots @hseelawi ... I think the old file format will work in our case.
Thanks for sharing these files with us. Just to confirm these are the same data but different format?
Just on other thing to mention before closing the issue. 
I think you have a bug in score calculation of FID:

In the [notebook](https://github.com/Alue-Benchmark/alue_baselines/blob/master/notebooks/USE/IDAT-irony-detection.ipynb) you are using the `average="macro"` argument while in [compute_metrics.py](https://github.com/Alue-Benchmark/alue_baselines/blob/master/bert-baselines/compute_metrics.py) you are using the default `average="binary"`

These gives 2 different scores, so I am confused which one is correct 

Thanks

**Comment by hseelawi** [comment]
This should be the same data but a different format indeed.

Regarding the FID task, we use the macro average as in the notebook and the original competition, but some of our scripts were experimental in nature so they might be slightly modified. We are sorry for any confusion caused.

**Comment by ghaddarAbs** [comment]
Thanks for the clarifications @hseelawi !!! They were very useful
I will close the issue the keep an eye on any update of the code.
However, our model is in the last phase of pretraining and we are expecting to submit to the leaderboard next week.

Best, 
Abbas

**Comment by Jason3900** [comment]
> That would in principle work but since we have to abide by the licence terms of the original authors we should first obtain some form of authorization. This might take a while but nonetheless, Dr. Habash has agreed that we send you the original data for now. I will send it over via the google group thread for this issue. Also, kindly notice that I have caught a bug in the DataGenerator class in the notebook. It seems that we are using an outdated version of it. I will be updating this very soon. Please feel free to reply or close this issue.
> 
> Kind regards, Haitham

Hey! Is it possible that you share the MDD dataset you've merged? I've got an original copy, but I need the correct order to combine, too.

**Comment by hseelawi** [comment]
Hi Jason,

I believe I sent you the data already. I will also attach it to this email
anyway for your convenience.

Please let me know if you have any questions or concerns.

Kind regards,
Haitham

**Comment by Jason3900** [comment]
Thanks a lot, I've got your email and data.

Haitham Seelawi ***@***.***> 于2022年7月19日周二 15:00写道：

> Hi Jason,
>
> I believe I sent you the data already. I will also attach it to this email
> anyway for your convenience.
>
> Please let me know if you have any questions or concerns.
>
> Kind regards,
> Haitham
>

**Comment by Jason3900** [comment]
Hey, thanks for sharing the data. However, I found that the the submission results on MDD is not the same as the ones I run locally. How does that happen?
